### PR TITLE
modules: mbedtls: introduce MBEDTLS_MEMORY_DEBUG option

### PR DESCRIPTION
**Pull request by mniestroj**
This option allows enable mbed TLS debugging of buffer allocator memory
issues.

There is a mbedtls reporitory PR that select proper macro here: https://github.com/zephyrproject-rtos/mbedtls/pull/17

### CODE REVIEW

**Comment by mniestroj** [comment]
Please note that prints on stderr (which are described in Kconfig help message) do not work with minimal libc, but they do with newlib. There is a PR to have stderr on console: #29821.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
LGTM.

BTW, is there still extra `\n` after each mbedtls debug print by logger? I remember seeing some mbedtls debug output and it was hard to follow because of those extra newlines.

**Comment by mniestroj** [comment]
> BTW, is there still extra `\n` after each mbedtls debug print by logger? I remember seeing some mbedtls debug output and it was hard to follow because of those extra newlines.

Yes, there is `\n` on the end of each `mbedtls_fprintf()`, which resolves right now to `fprintf()`. But this is different than `mbedtls_debug_print_msg()`, which resolves to `NET_DBG()` if `CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=y` (I guess this is what you had in mind).

**Comment by jukkar** [comment]
> But this is different than `mbedtls_debug_print_msg()`, which resolves to `NET_DBG()` if `CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=y` (I guess this is what you had in mind).

It was couple of months ago that I saw this and it looked like a bug but I did not start to investigate it further.

File: west.yml [hunk 1]
```diff
@@ -90,7 +90,7 @@ manifest:
       revision: 9339a1331cc36e82c180d8277ab22953282595d9
       path: modules/lib/gui/lvgl
     - name: mbedtls
-      revision: aef137b1af8aa7a0f43345c82459254b8832262e
+      revision: pull/17/head
```
Comment (MaureenHelm) [inline west.yml:93]: The module PR is now merged, so you can update this to the new SHA

**Review by MaureenHelm — APPROVED** [review]
